src/components/QuestionDisplay/QuestionDisplay.jsx

```diff
@@ -55,7 +54,7 @@ function QuestionDisplay() {
           <Box sx={{ display: 'flex', flexDirection: 'column', gap: 1 }}>
             {decodedShuffledAnswers.map((decodedAnswer, index) => (
               <AnswerButton
-                key={uuidv4()}
+                key={`${currentQuestion.id}-answer-${index}`}
```

Comment:
Why do you need to use index here? Just use the id and you are good to go. Using indexes as keys is always a bad idea, even when you use it partially.